internal/resources/metal/virtual_circuit/resource.go

```diff
@@ -145,95 +172,104 @@ func Resource() *schema.Resource {
-	name := d.Get("name").(string)
-
-	if _, ok := d.GetOk("vlan_id"); ok {
-		vncr.VlanVirtualCircuitCreateInput = &metalv1.VlanVirtualCircuitCreateInput{
-			ProjectId:   projectId,
-			Name:        &name,
-			Description: metalv1.PtrString(d.Get("description").(string)),
-			Speed:       metalv1.PtrString(d.Get("speed").(string)),
-			Vnid:        metalv1.PtrString(d.Get("vlan_id").(string)),
-		}
+	if _, ok := d.GetOk("virtual_circuit_id"); ok {
+		vcId := d.Get("virtual_circuit_id").(string)
+		d.SetId(vcId)
+		return resourceMetalVirtualCircuitUpdate(ctx, d, meta)
 	} else {
-		vncr.VrfVirtualCircuitCreateInput = &metalv1.VrfVirtualCircuitCreateInput{
-			ProjectId:   projectId,
-			Name:        &name,
-			Description: metalv1.PtrString(d.Get("description").(string)),
-			Speed:       metalv1.PtrString(d.Get("speed").(string)),
-			Vrf:         d.Get("vrf_id").(string),
-			// TODO: woof
-			Md5:        *metalv1.NewNullableString(metalv1.PtrString(d.Get("md5").(string))),
-			PeerAsn:    int64(d.Get("peer_asn").(int)),
-			Subnet:     d.Get("subnet").(string),
-			CustomerIp: metalv1.PtrString(d.Get("customer_ip").(string)),
-			MetalIp:    metalv1.PtrString(d.Get("metal_ip").(string)),
-		}
-	}
+		client := meta.(*config.Config).NewMetalClientForSDK(d)
+		vncr := metalv1.VirtualCircuitCreateInput{}
+
+		connId := d.Get("connection_id").(string)
+		portId := d.Get("port_id").(string)
+		projectId := d.Get("project_id").(string)
+		name := d.Get("name").(string)
 
-	tags := d.Get("tags.#").(int)
-	if tags > 0 {
 		if _, ok := d.GetOk("vlan_id"); ok {
-			vncr.VlanVirtualCircuitCreateInput.Tags = converters.IfArrToStringArr(d.Get("tags").([]interface{}))
+			vncr.VlanVirtualCircuitCreateInput = &metalv1.VlanVirtualCircuitCreateInput{
+				ProjectId:   projectId,
+				Name:        &name,
+				Description: metalv1.PtrString(d.Get("description").(string)),
+				Speed:       metalv1.PtrString(d.Get("speed").(string)),
+				Vnid:        metalv1.PtrString(d.Get("vlan_id").(string)),
+			}
 		} else {
-			vncr.VrfVirtualCircuitCreateInput.Tags = converters.IfArrToStringArr(d.Get("tags").([]interface{}))
+			vncr.VrfVirtualCircuitCreateInput = &metalv1.VrfVirtualCircuitCreateInput{
+				ProjectId:   projectId,
+				Name:        &name,
+				Description: metalv1.PtrString(d.Get("description").(string)),
+				Speed:       metalv1.PtrString(d.Get("speed").(string)),
+				Vrf:         d.Get("vrf_id").(string),
+				// TODO: woof
+				Md5:          *metalv1.NewNullableString(metalv1.PtrString(d.Get("md5").(string))),
+				PeerAsn:      int64(d.Get("peer_asn").(int)),
+				Subnet:       d.Get("subnet").(string),
+				CustomerIp:   metalv1.PtrString(d.Get("customer_ip").(string)),
+				MetalIp:      metalv1.PtrString(d.Get("metal_ip").(string)),
+				SubnetIpv6:   metalv1.PtrString(d.Get("subnet_ipv6").(string)),
+				CustomerIpv6: metalv1.PtrString(d.Get("customer_ipv6").(string)),
+				MetalIpv6:    metalv1.PtrString(d.Get("metal_ipv6").(string)),
```

Comment:
If the user doesn't specify values for these attributes, this code will send, e.g., `metal_ip: ""`.  Does the API accept an empty string for these parameters or will it return an error in that case?

Reply:
It will accept an empty string and calculate those fields itself the same as if the parameters were not sent.